ios/RCTWKWebView/RCTWKWebView.m

```diff
@@ -138,7 +138,7 @@ - (void)resetupScripts {
 
 - (void)setupPostMessageScript {
   if (_messagingEnabled) {
-    NSString *source=@"window.originalPostMessage = window.postMessage; window.postMessage = function (data) { window.webkit.messageHandlers.reactNative.postMessage(data); }";
+    NSString *source=@"window.originalPostMessage = window.postMessage; window.postMessage = function(message, targetOrigin, transfer) { return typeof targetOrigin === 'undefined' ? window.webkit.messageHandlers.reactNative.postMessage(message) : window.originalPostMessage(message, targetOrigin, transfer); };";
```

Comment:
I wonder if instead the following would take care of all the concerns in the discussion:

```
window.originalPostMessage = window.postMessage; 
window.postMessage = function(message, targetOrigin, transfer) { 
  window.webkit.messageHandlers.reactNative.postMessage(message);
  if (typeof targetOrigin !== 'undefined') { 
    window.originalPostMessage(message, targetOrigin, transfer); 
  }
};
```

We forward the message to RN listeners, and if it's eligible, also to existing listeners.  By eligible I mean it has enough arguments not to break the native call with an error.

I think the root of the problems is that if we _replace_ code that was there to handle posted messages, all existing clients will break, because their messages will not be delivered in the way they expect.  The above code continues to deliver them as expected, plus also delivers them to our RN listeners.

This means that several listeners are getting messages off this public message bus that they will not be interested in, but that's already the case for all the listeners: they're already obliged to filter messages for common sense if they're to function correctly.

Reply:
In the code in the PR, there's provision for a return value from `postMessage`, which my snippet does not provide, but I'm not sure what the return value is supposed to be in any case?  I can't see where that's documented, and don't know that any users of `postMessage` will be expecting a return value?

Reply:
I still prefer a more explicit approach. But this solutions seems to work for most of the use cases. I merge this in 2b3d862bc892719aa0440ef3cb69145124eb2046